packages/backend-api/install/tests/ShopBundle/Smoke/BackendApiCreateProductTest.php

```diff
@@ -105,56 +117,65 @@ public function testCreateEmptyProduct(): void
 
         $uuid = $this->extractUuid($location);
 
+        $nullsByLocale = [];
+        $nullsByDomainId = [];
+        foreach ($this->domain->getAll() as $domainConfig) {
+            $nullsByLocale[$domainConfig->getLocale()] = null;
+            $nullsByDomainId[$domainConfig->getId()] = null;
+        }
+
         $expected = [
             'uuid' => $uuid,
-            'name' => [
-                'en' => null,
-                'cs' => null,
-            ],
+            'name' => $nullsByLocale,
             'hidden' => false,
             'sellingDenied' => false,
             'sellingFrom' => null,
             'sellingTo' => null,
             'catnum' => null,
             'ean' => null,
             'partno' => null,
-            'shortDescription' => [
-                1 => null,
-                2 => null,
-            ],
-            'longDescription' => [
-                1 => null,
-                2 => null,
-            ],
+            'shortDescription' => $nullsByDomainId,
+            'longDescription' => $nullsByDomainId,
         ];
         $actualProduct = $this->getProduct($uuid);
         $this->assertEquals($expected, $actualProduct);
     }
 
     public function testValidationError(): void
     {
+        $namesByLocale = [];
+        $shortDescriptionsByDomainId = [];
+        $longDescriptionsByDomainId = [];
+        $firstDomainLocale = $this->domain->getDomainConfigById(1)->getLocale();
+        $notExistingDomainId = $this->getNotExistingDomainId();
+        foreach ($this->domain->getAll() as $domainConfig) {
+            $domainLocale = $domainConfig->getLocale();
+            $domainId = $domainConfig->getId();
+            if ($domainId === 1) {
```

Comment:
Well, we set expected fields that will fail the validation for the first domain and then we are asserting validation error message in this test. We need to have set something for the other domains as well, otherwise, the validation messages do not fit.

Reply:
Ok, that make sense since you do not want to test valiadation error for both locales.

Maybe you could rename testing method to `testUnexpectedProductTranslationFails`